src/controllers/continuousUpdates/continuousUpdates.ts

```diff
@@ -0,0 +1,206 @@
+
+/* eslint-disable @typescript-eslint/no-floating-promises */
+
+const getIntervalRefreshTime = (constUpdateInterval: number, newestOpTimestamp: number): number => {
+  // 5s + new Date().getTime() - timestamp of newest op / 10
+  // here are some example of what this means:
+  // 1s diff between now and newestOpTimestamp: 5.1s
+  // 10s diff between now and newestOpTimestamp: 6s
+  // 60s diff between now and newestOpTimestamp: 11s
+  // 5m diff between now and newestOpTimestamp: 35s
+  // 10m diff between now and newestOpTimestamp: 65s
+  return newestOpTimestamp === 0
+    ? constUpdateInterval
+    : constUpdateInterval + (new Date().getTime() - newestOpTimestamp) / 10
+}
+export class ContinuousUpdatesController {
+  #main: IMainController
+
+  updatePortfolioInterval: RecurringTimeout
+
+  accountsOpsStatusesInterval: RecurringTimeout
+
+  accountStateLatestInterval: RecurringTimeout
+
+  accountStatePendingInterval: RecurringTimeout
+
+  fastAccountStateReFetchTimeout: RecurringTimeout
+
+  #retriedFastAccountStateReFetchForNetworks: string[] = []
+
+  constructor({ main }: { main: IMainController }) {
+    this.#main = main
+
+    // Postpone the portfolio update for the next interval
+    // if we have broadcasted but not yet confirmed acc op.
+    // Here's why:
+    // 1. On the Dashboard, we show a pending-to-be-confirmed token badge
+    //    if an acc op has been broadcasted but is still unconfirmed.
+    // 2. To display the expected balance change, we calculate it from the portfolio's pending simulation state.
+    // 3. When we sign and broadcast the acc op, we remove it from the Main controller.
+    // 4. If we trigger a portfolio update at this point, we will lose the pending simulation state.
+    // 5. Therefore, to ensure the badge is displayed, we pause the portfolio update temporarily.
+    //    Once the acc op is confirmed or failed, the portfolio interval will resume as normal.
+    // 6. Gotcha: If the user forcefully updates the portfolio, we will also lose the simulation.
+    //    However, this is not a frequent case, and we can make a compromise here.
+    this.updatePortfolioInterval = createRecurringTimeout(async () => {
+      if (this.#main.activity.broadcastedButNotConfirmed.length) return
+      await this.#main.updateSelectedAccountPortfolio()
+    }, INACTIVE_EXTENSION_PORTFOLIO_UPDATE_INTERVAL)
+
+    this.#main.ui.uiEvent.on('addView', () => {
+      if (this.#main.ui.views.length === 1) {
+        this.updatePortfolioInterval.restart({
+          timeout: ACTIVE_EXTENSION_PORTFOLIO_UPDATE_INTERVAL
+        })
+        this.fastAccountStateReFetchTimeout.start()
+      }
+    })
+    this.#main.ui.uiEvent.on('removeView', () => {
+      if (!this.#main.ui.views.length) {
+        this.updatePortfolioInterval.restart({
+          timeout: INACTIVE_EXTENSION_PORTFOLIO_UPDATE_INTERVAL
+        })
+        this.fastAccountStateReFetchTimeout.stop()
+      }
+    })
+
```

Comment:
We shouldn't start/stop fastAccountStateReFetchTimeout on these events. Only on provider controller update